### ISSUE TITLE
[BUG] Exceptions without 'spaces' in their Value take long to print.

### ISSUE THREAD

**Issue by TyberiusPrime** [willmcgugan/rich#947]
**Describe the bug**
If you have an exception that has no 'natural line breaks'
and it outputs a large amount of text,
the pretty traceback will take a surprisingly long amount of time.

**To Reproduce**
```python
import rich.traceback
rich.traceback.install()
raise ValueError("h" * 50000)
```
Takes 22 seconds, 
while "h " * 25000 takes 800 ms.

**Platform**
Ubuntu 20.10, kitty, current github (pip freeze reports 9.8.2)

Suggestings: We either improve the performance when forcing line breaks,
or we somehow truncate the exception's output to a sane value.

Yes this is an edge case.
Yes I have run into it.

**Comment by willmcgugan** [comment]
I should be able to fix that, I narrowed it down to one pathological regex

**Comment by willmcgugan** [comment]
Thanks, fixed in v9.9.9
